src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs

```diff
@@ -47,17 +51,53 @@ public void Start(PushRuntimeSettings limitations)
 
             if (purgeOnStartup)
             {
-                Array.ForEach(Directory.GetFiles(path), File.Delete);
+                PurgeDirectories();
             }
 
+            CreateDirectories();
+
+            RecoverPendingTransactions();
+
+            messagePumpTask = Task.Run(ProcessMessages, cancellationToken);
+
+            delayedMessagePoller.Start();
+        }
+
+        void RecoverPendingTransactions()
+        {
             if (transactionMode != TransportTransactionMode.None)
             {
-                DirectoryBasedTransaction.RecoverPartiallyCompletedTransactions(path);
+                DirectoryBasedTransaction.RecoverPartiallyCompletedTransactions(path, PendingDirName, CommittedDirName);
```

Comment:
do we really need to delete the pending and commited directories when recovering leftover transactions? otherwise we could remove the next line...